openhtf/__init__.py

```diff
@@ -45,6 +46,106 @@ class InvalidTestPhaseError(Exception):
   """Raised when an invalid method is decorated."""
 
 
+def Test(*phases, **metadata):
+  return TestInfo(phases, metadata=metadata)
+
+
+class TestInfo(mutablerecords.Record(
```

Comment:
I make `TestInfo.Execute()` call `ExecuteTest()` which makes it more like a record with a helper function. The point of all this is to make the `TestInfo` class become stateless and separate execution from initialization, that was my proposal to the openhtf meeting and that was accepted. If you disagree, I'd rather we talk about that than this detail